src/Functions/generateUUIDv7.cpp

```diff
@@ -0,0 +1,351 @@
+{
+public:
+    explicit FunctionGenerateUUIDv7(ContextPtr context) : selector(context)
+    {
+        selector.registerImplementation<TargetArch::Default, TargetSpecific::Default::FunctionGenerateUUIDv7>();
+
+#if USE_MULTITARGET_CODE
+        selector.registerImplementation<TargetArch::AVX2, TargetSpecific::AVX2::FunctionGenerateUUIDv7>();
+#endif
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t input_rows_count) const override
+    {
+        return selector.selectAndExecute(arguments, result_type, input_rows_count);
+    }
+
+    static FunctionPtr create(ContextPtr context) { return std::make_shared<FunctionGenerateUUIDv7>(context); }
+
+private:
+    ImplementationSelector<IFunction> selector;
+};
+
+class FunctionGenerateUUIDv7WithCounter : public TargetSpecific::Default::FunctionGenerateUUIDv7WithCounter
+{
+public:
+    explicit FunctionGenerateUUIDv7WithCounter(ContextPtr context) : selector(context)
+    {
+        selector.registerImplementation<TargetArch::Default, TargetSpecific::Default::FunctionGenerateUUIDv7WithCounter>();
+
+#if USE_MULTITARGET_CODE
+        selector.registerImplementation<TargetArch::AVX2, TargetSpecific::AVX2::FunctionGenerateUUIDv7WithCounter>();
+#endif
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t input_rows_count) const override
+    {
+        return selector.selectAndExecute(arguments, result_type, input_rows_count);
+    }
+
+    static FunctionPtr create(ContextPtr context) { return std::make_shared<FunctionGenerateUUIDv7WithCounter>(context); }
+
+private:
+    ImplementationSelector<IFunction> selector;
+};
+
+
+class FunctionGenerateUUIDv7WithFastCounter : public TargetSpecific::Default::FunctionGenerateUUIDv7WithFastCounter
+{
+public:
+    explicit FunctionGenerateUUIDv7WithFastCounter(ContextPtr context) : selector(context)
+    {
+        selector.registerImplementation<TargetArch::Default, TargetSpecific::Default::FunctionGenerateUUIDv7WithFastCounter>();
+
+#if USE_MULTITARGET_CODE
+        selector.registerImplementation<TargetArch::AVX2, TargetSpecific::AVX2::FunctionGenerateUUIDv7WithFastCounter>();
+#endif
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t input_rows_count) const override
+    {
+        return selector.selectAndExecute(arguments, result_type, input_rows_count);
+    }
+
+    static FunctionPtr create(ContextPtr context) { return std::make_shared<FunctionGenerateUUIDv7WithFastCounter>(context); }
+
+private:
+    ImplementationSelector<IFunction> selector;
+};
+
+
+REGISTER_FUNCTION(GenerateUUIDv7)
+{
+    factory.registerFunction<FunctionGenerateUUIDv7>(
+        FunctionDocumentation{
+            .description = R"(
+Generates a UUID of version 7 with current Unix time having milliseconds precision followed by random data.
```

Comment:
The syntax for in-source docs is generally not pretty. Defining the individual components as separate variables may be a bit simpler and allow re-use of documentation between the three functions, see src/Functions/locate.cpp.

Reply:
(I changed the docs a bit but did not update this in-source docs, it would be great if you could do that.)

Reply:
> The syntax for in-source docs is generally not pretty. Defining the individual components as separate variables may be a bit simpler and allow re-use of documentation between the three functions, see src/Functions/locate.cpp.

I've just found a mistype in the [`doc_description`](https://github.com/ClickHouse/ClickHouse/blob/ab29a5f428b4690cc1602766e00ea6ef1df1e6b6/src/Functions/locate.cpp#L22): "The previous behaviorcan  be restored"

It looks like in-source docs need to be dumped to some text file during CI and spell-checked as the ones in the docs folder.